mangaki/mangaki/views.py

```diff
@@ -614,20 +614,23 @@ def get_works(request, category):
     return HttpResponse(json.dumps(data), content_type='application/json')
 
 
-def get_reco_list(request, category, editor):
+@login_required
+def get_reco_algo_list(request, algo, category):
     reco_list = []
-    for work, is_manga, in_willsee in get_recommendations(request.user, category, editor):
-        reco_list.append({'id': work.id, 'title': work.title, 'poster': work.ext_poster, 'synopsis': work.synopsis,
-            'category': 'manga' if is_manga else 'anime', 'rating': 'willsee' if in_willsee else None})
-    return HttpResponse(json.dumps(reco_list), content_type='application/json')
-
+    data = get_reco_algo(request.user, algo, category)
+    works = data['works']
+    for work_id in data['work_ids']:
+        work = works[work_id]
+        reco_list.append({'id': work.id, 'title': work.title, 'poster': work.ext_poster, 'synopsis': work.synopsis, 'category': work.category.slug})
+    return HttpResponse(json.dumps(reco_list, indent=4), content_type='application/json')
```

Comment:
Alright, I didn't mean you to use `serializers.serialize(Work, …)`, but just `serializers.serialize(reco_list, …)` ; just don't forget that Django can handle datetimes, decimals, promises and UUID automagically. Though, I don't know if `json.dumps` is faster than Django serializer (but, I think that the backend is `json.dumps`, _but_, it can be changed through modules or whatever.)